sympy/stats/crv.py

```diff
@@ -417,7 +417,7 @@ def moment_generating_function(self, t, **kwargs):
 
     def expectation(self, expr, var, evaluate=True, **kwargs):
         """ Expectation of expression over distribution """
-        if evaluate:
+        if evaluate and expr:
```

Comment:
Testing `if expr` like this will not catch all cases e.g.:
```python
In [4]: e = x*(x + 1) - x*x - x

In [5]: e
Out[5]: 
   2                
- x  + x⋅(x + 1) - x

In [6]: bool(e)
Out[6]: True

In [7]: expand(e)
Out[7]: 0

In [8]: Poly(e, x)
Out[8]: Poly(0, x, domain='ZZ')

In [9]: Poly(e, x).degree()
Out[9]: -∞
```
If would be better to test if `p` is zero.

A test for a case like this should be added.